### ISSUE TITLE
How to compile it with libde265 build in

### ISSUE THREAD

**Issue by lishan-2h** [strukturag/libheif#730]
First I use command:
_mkdir build
cd build
cmake .. -D LIBDE265_INCLUDE_DIR=../../libde265/build -D LIBDE265_LIBRARY=../../libde265
make_
To compile it and error log is :
```
/mnt/d/Code/libheif/libheif/plugins/heif_decoder_libde265.cc:35:10: fatal error: libde265/de265.h: No such file or directory
   35 | #include <libde265/de265.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [libheif/CMakeFiles/heif.dir/build.make:277: libheif/CMakeFiles/heif.dir/plugins/heif_decoder_libde265.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:310: libheif/CMakeFiles/heif.dir/all] Error 2
```
so I change cmake command like:
_cmake .. -D LIBDE265_INCLUDE_DIR=../../libde265 -D LIBDE265_LIBRARY=../../libde265/build_
And error log is :

```
_In file included from /mnt/d/Code/libheif/libheif/plugins/heif_decoder_libde265.cc:35:
/mnt/d/Code/libde265/libde265/de265.h:29:10: fatal error: libde265/de265-version.h: No such file or directory
   29 | #include <libde265/de265-version.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[2]: *** [libheif/CMakeFiles/heif.dir/build.make:277: libheif/CMakeFiles/heif.dir/plugins/heif_decoder_libde265.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:310: libheif/CMakeFiles/heif.dir/all] Error 2
make: *** [Makefile:171: all] Error 2_
```

And it seems **de265.h** and **de265-version.h** are in different directory. So I don't know how to config LIBDE265_INCLUDE_DIR and LIBDE265_LIBRARY correctly. Did anyone know it? I would appreciate it if someone could tell me.  Thanks.

**Comment by silverbacknet** [comment]
Have you actually built libde265 successfully? CMake generates the version file automatically.

**Comment by bigcat88** [comment]
> cmake .. -D LIBDE265_INCLUDE_DIR=../../libde265/build -D LIBDE265_LIBRARY=../../libde265

from that `build` folder you created:
`cmake -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_BUILD_TYPE=Release ..`
is enough for `libde265`.

Did you install `libde265` after build? Or do you want to link it statically to `libheif` as a single file?

**Comment by lishan-2h** [comment]
> Have you actually built libde265 successfully? CMake generates the version file automatically.

Yes, I have build libde265, and it generate version file in `~/libde265/build/libde265/de265-version.h` 
But it seems need both `de265-version.h` and `de265.h` to comile libheif with libde265.  
If set `LIBDE265_INCLUDE_DIR = ~/libde265/build`, compiler will report error that cannot find `de265.h` . Because `de265.h` was in `~/libde265/libde265`.

**Comment by lishan-2h** [comment]
> 

I didn't install `libde265` after build, because I don't have administrator permissions.
Yes, I just want to link it statically to libheif as a single file.

**Comment by silverbacknet** [comment]
That's because `de265-version.h` is supposed to be in the source/header folder, not the build folder. No idea how it could have ended up in there or how it built that way.

**Comment by farindk** [comment]
`de265-version.h` is not in the source folder because it is generated during the build process.

@lishan-2h: You have to install the library so that all files are at the correct position. I understand that you have no administrator rights, but you can install it into your local directory (`CMAKE_INSTALL_PREFIX=...`). Then set the required paths accordingly (`LD_LIBRARY_PATH` and `PKG_CONFIG_PATH`).
